### PR TITLE
Request mp4/hls src object for owned vimeo videos

### PR DESCRIPTION
**Pull request by uxengine**
Related PRs:
https://github.com/orbiting/styleguide/pull/60
https://github.com/orbiting/publikator-backend/pull/21

### CODE REVIEW

File: components/editor/modules/embed/EmbedLoader.js [hunk 1]
```diff
@@ -70,6 +70,10 @@ export default (query, Component) => (
       const active = editor.value.blocks.some(
         block => block.key === node.key
       )
+
+      const data = node.data.toJS()
+      // We need to guard against undefined thumbnail below which happens
```
Comment (uxengine) [inline components/editor/modules/embed/EmbedLoader.js]: Note that I still need this even with
https://github.com/orbiting/styleguide/commit/3590bfd53efdd0e3ffa5e01660e086ff9ca01204

File: components/editor/modules/embed/EmbedLoader.js [hunk 2]
```diff
@@ -79,9 +83,7 @@ export default (query, Component) => (
               data-active={active}
               contentEditable={false}
               >
-              <Component
-                data={node.data.toJS()}
-                />
+              {data.thumbnail && <Component data={data} />}
```
Comment (uxengine) [inline components/editor/modules/embed/EmbedLoader.js]: Note that I still need this even with
https://github.com/orbiting/styleguide/commit/3590bfd53efdd0e3ffa5e01660e086ff9ca01204

File: components/editor/modules/embed/EmbedLoader.js (hunk 1 above)
Comment (tpreusse) [inline components/editor/modules/embed/EmbedLoader.js]: let's fix the root issue: the loading state is broken. Which is also responsible for quick flashes of NaN dated tweets in our demo.

In the constructor `loading: false` is bs—should be `loading: !node.data.has('id'), error: !node.data.has('id') && !node.data.has('queryParams') && 'No embed params found.'` (props and therefore node is available as the first arg of the constructor)

Comment (uxengine) [reply components/editor/modules/embed/EmbedLoader.js]: Done as just paired.

**Pull request merged**